PWGHF/HFC/Tasks/taskCorrelationDMesonPairs.cxx

```diff
@@ -51,60 +51,41 @@ enum D0Type {
 };
 } // namespace
 
-// string definitions, used for histogram axis labels
-const char stringCorrelationPairs[193] = "D meson pair candidates 2D;inv. mass D_{1} (GeV/#it{c}^{2});inv. mass D_{2} (GeV/#it{c}^{2});#it{p}_{T}^{D_{1}} (GeV/#it{c});#it{p}_{T}^{D_{2}} (GeV/#it{c});#eta D_{1};#eta D_{2};type1;type2;";
-const char stringCorrelationPairsFinerBinning[207] = "D meson pair candidates 2D Finer Binning;inv. mass D_{1} (GeV/#it{c}^{2});inv. mass D_{2} (GeV/#it{c}^{2});#it{p}_{T}^{D_{1}} (GeV/#it{c});#it{p}_{T}^{D_{2}} (GeV/#it{c});#eta D_{1};#eta D_{2};type1;type2;";
-
-// definition of vectors for standard ptbin and invariant mass configurables
-const int nPtBinsCorrelations = 8;
-const double ptBinsCorrelations[nPtBinsCorrelations + 1] = {0., 2., 4., 6., 8., 12., 16., 24., 99.};
-auto vecPtBinsCorrelations = std::vector<double>{ptBinsCorrelations, ptBinsCorrelations + nPtBinsCorrelations + 1};
-
 struct HfTaskCorrelationDMesonPairs {
-  // Enable histograms with finer pT and y binning
-  Configurable<bool> enableFinerBinning{"enableFinerBinning", false, "Enable histograms with finer pT and y binning"};
-  Configurable<std::vector<double>> binsPtCorrelations{"binsPtCorrelations", std::vector<double>{vecPtBinsCorrelations}, "pT bin limits for correlation plots"};
-
-  // HistoTypes
-  HistogramConfigSpec hTHnMass2DCorrPairs{HistType::kTHnSparseD, {{200, 1.6, 2.1}, {200, 1.6, 2.1}, {10, 0., 10.}, {10, 0., 10.}, {10, -1, 1}, {10, -1, 1}, {4, -0.5, 3.5}, {4, -0.5, 3.5}}}; // note: axes 3 and 4 (the pT) are updated in the init();
-  HistogramConfigSpec hTHnMass2DCorrPairsFinerBinning{HistType::kTHnSparseD, {{200, 1.6, 2.1}, {200, 1.6, 2.1}, {60, 1., 6.}, {60, 1., 6.}, {160, -0.8, 0.8}, {160, -0.8, 0.8}, {4, -0.5, 3.5}, {4, -0.5, 3.5}}};
+  // Configurables to set sparse axes
+  ConfigurableAxis thnConfigAxisInvMassCand1{"thnConfigAxisInvMassCand1", {200, 1.6, 2.1}, "Cand. 1 Inv-mass bins"};
+  ConfigurableAxis thnConfigAxisInvMassCand2{"thnConfigAxisInvMassCand2", {200, 1.6, 2.1}, "Cand. 2 Inv-mass bins"};
+  ConfigurableAxis thnConfigAxisPtCand1{"thnConfigAxisPtCand1", {25, 0., 24.}, "Cand. 1 pT bins"};
+  ConfigurableAxis thnConfigAxisPtCand2{"thnConfigAxisPtCand2", {25, 0., 24.}, "Cand. 2 pT bins"};
+  ConfigurableAxis thnConfigAxisYCand1{"thnConfigAxisYCand1", {10, -1, 1}, "Cand. 1 rapidity bins"};
+  ConfigurableAxis thnConfigAxisYCand2{"thnConfigAxisYCand2", {10, -1, 1}, "Cand. 2 rapidity bins"};
+  ConfigurableAxis thnConfigAxisCandType1{"thnConfigAxisCandType1", {4, -0.5, 3.5}, "Cand. 1 Type"};
+  ConfigurableAxis thnConfigAxisCandType2{"thnConfigAxisCandType2", {4, -0.5, 3.5}, "Cand. 2 Type"};
```

Comment:
Could these be reduced to one configurable per variable for both candidates (e.g. only one for the mass, one for the pT, one for the y and one for the candidate type)? Or do you think that one could have different binning for the two candidates?